position_in_partition.hh

```diff
@@ -194,6 +194,10 @@ public:
         return bound_view(*_ck, _bound_weight == bound_weight::before_all_prefixed ? bound_kind::excl_end : bound_kind::incl_end);
     }
 
+    bool adjacent(const schema& s, position_in_partition_view o) const {
+        return as_end_bound_view().adjacent(s, o.as_start_bound_view());
```

Comment:
Note that for position ranges, adjacent() is the same as equality. [x, y) is adjacent to [z, w) if y == z.

Reply:
I'm confused. Are position ranges always `[)` (closed start bound, open end bound)?

Reply:
Yes, in the domain of `position_in_partition`, they're always `[a, b)`. The end p_i_p (`b`) can be still either before or after a key, in which case the range either includes the key or it excludes the key. If you have ranges `[a, before(x))` and `[before(x), c)`, such ranges are adjacent, because all elements in the range `[a, c)` belong to either one or the other.